src/assets/CollectibleDetectionController.ts

```diff
@@ -145,12 +145,20 @@ export class CollectibleDetectionController extends BaseController<
       /* istanbul ignore if */
       do {
         const api = this.getOwnerCollectiblesApi(address, offset);
-        response = await timeoutFetch(
-          api,
-          openSeaApiKey ? { headers: { 'X-API-KEY': openSeaApiKey } } : {},
-          15000,
-        );
-        const collectiblesArray = await response.json();
+        try {
+          response = await timeoutFetch(api, {}, 15000);
+        } catch {
```

Comment:
Would it be possible to throw this and then we can prompt the user before we decide to use the OpenSea API directly? If they decided they don't need the data I don't want to force an OpenSea API call without user consent.

Reply:
We won't ever even hit this point unless the user has turned on the OpenSea API. I don't know if we need to surface a difference to the user between calling our proxy of OpenSea and OpenSea directly?

Reply:
The controller won't detect at all while `disabled` is true. It is set to [true by default](https://github.com/MetaMask/controllers/blob/main/src/assets/CollectibleDetectionController.ts#L220) and only flip false if the user [has opted to turn on preferences for OpenSea API and NFT AutoDetection](https://github.com/MetaMask/controllers/blob/main/src/assets/CollectibleDetectionController.ts#L232)